app-frontend/src/app/components/histogram/dataVizHistogram/index.js

```diff
@@ -0,0 +1,114 @@
+    }
+
+    $postLink() {
+        this.isLoadingHistogram = true;
+        this.$q.all({
+            histStat: this.histogram,
+            histBounds: this.bounds
+        }).then(({histStat, histBounds}) => {
+            const {histogram, statistics} = histStat;
+            if (histogram && histBounds && get(histogram, 'buckets.length')) {
+                this.$scope.$evalAsync(() => this.initGraph(histogram, histBounds));
+                this.getGraph().then(graph => {
+                    this.createPlotFromHistogram(histogram, histBounds, statistics);
+                    graph.setData({histogram: this.plot})
+                        .render();
+                    this.isLoadingHistogram = false;
+                });
+            }
+        });
+    }
+
+    initGraph(histogram, bounds) {
+        this.options = {
+            dataRange: {
+                min: bounds.min,
+                max: bounds.max
+            },
+            height: 50,
+            style: {
+                height: '50px'
+            },
+            type: 'stat',
+            margin: {
+                top: 0,
+                bottom: 0,
+                right: 0,
+                left: 0
+            }
+        };
+
+        const elem = this.$element.find('.graph-container svg')[0];
+        this.graph = this.graphService.register(elem, this.id, this.options);
+    }
+
+    getGraph() {
+        return this.graphService.getGraph(this.id);
+    }
+
+    createPlotFromHistogram(histogram, bounds, statistics) {
+        const diff = (bounds.max - bounds.min) / 100;
+        const magnitude = Math.round(Math.log10(diff));
+        this.precision = Math.pow(10, magnitude);
+
+        const buckets = histogram.buckets;
+        let plot;
+        if (buckets.length < 100) {
+            let valueMap = new Map();
+            buckets.forEach(([bucket, value]) => {
+                const roundedBucket = Math.round(bucket / this.precision) * this.precision;
+                valueMap = valueMap.set(roundedBucket, value);
+            });
+            range(0, 100).forEach((mult) => {
+                const key = Math.round(
+                    (mult * diff + this.options.dataRange.min) / this.precision
+                ) * this.precision;
+                const value = valueMap.get(key, 0);
+                valueMap = valueMap.set(key, value);
+            });
+
+            plot = valueMap.entrySeq()
+                .sort(([K1], [K2]) => K1 - K2)
+                .map(([key, value]) => ({
+                    x: min([max([key, histogram.minimum]), histogram.maximum]),
+                    y: value}))
+                .toArray();
+        } else {
+            plot = buckets.map((bucket) => {
+                return {
+                    x: Math.round(bucket[0]), y: bucket[1]
+                };
+            });
+        }
+
+        this.plot = plot;
+    }
+}
+
+const component = {
+    bindings: {
+        id: '<',
+        histogram: '<',
```

Comment:
I'm starting to think that we need to be more explicit on whether a binding is a value or a promise that returns a value we care about. I'm writing some components which take promises as well, and it could definitely be confusing if we try to pass a normal value into one of these components and get a "Cannot read property 'then' of undefined" error or something.

I propose appending "Promise" or "P" to these bindings, similar to how we do stuff like append "Option" or "O" to options in scala. This also lets us bind to variables like `this.histogram` inside the component since the name isn't taken by the binding.